numba/core/types/abstract.py

```diff
@@ -57,14 +59,14 @@ def _intern(cls, inst):
             _typecache[wr] = wr
             return inst
 
-    def __call__(cls, *args, **kwargs):
+    def __call__(cls, *args: pt.Any, **kwargs: pt.Any) -> "Type":
```

Comment:
if you accept the suggestion about `NumbaTypeInst` then it would apply here and everywhere else where it says Type.

Reply:
Done.  I think I caught all uses of `Type`.